### ISSUE TITLE
jspm caches not invalidated when npm registry is changed, potentially causing Error: getaddrinfo ENOTFOUND

### ISSUE THREAD

**Issue by forestjohnsonilm** [jspm/jspm-cli#1157]
To Reproduce:

Configure jspm to use a private npm mirror.
Install npm:angular through that private npm mirror.
Make the private npm mirror unreachable (hosts file entry)
Configure jspm to use the public npm registry.
Make a new project and run jspm init, then try to install npm:angular.

Expected: the package will be installed from the public npm registry.
Actual: installation fails because it tries to look up files in the private npm registry. Error: getaddrinfo ENOTFOUND

run jspm cache-clear

Expected: you should not have to do this to be able to install packages.
Actual: now angular can be installed successfully

**Comment by guybedford** [comment]
Thanks!

**Comment by guybedford** [comment]
I'm not sure I follow this actually - there is no way the private npm registry would get a lookup if you've configured jspm not to use it. Also `jspm cc` clears no information associated with this lookup, so shouldn't have any affect here. Are you sure your fix wasn't clearing out your `.npmrc` file?

**Comment by guybedford** [comment]
Not sure I fully follow this scenario without further info.

**Comment by forestjohnsonilm** [comment]
I suppose I forgot to mention that, I did actually edit the .npmrc file, but that wasn't what fixed my problem. I had to run jspm cache-clear before I could update packages that had been previously downloaded when I was using the private NPM registry, which felt wrong. And the command line commanded me to create a github issue if I felt that way ;)

I'm sorry it's not easily reproduce-able. if I have time later I'll see if I can create a minimal reproduction of it.
